Jenkinsfile

```diff
@@ -1,31 +1,39 @@
 @Library('shared-libraries') _
 
 def getJava(){
-    if(env.JAVA_VERSION=="JAVA17"){
-        return "/home/builder/java/jdk-17.0.2"
-    }else if(env.JAVA_VERSION=="JAVA11"){
-        return "/home/builder/java/jdk-11.0.2"
-    }else if(env.JAVA_VERSION=="JAVA21"){
+	if(env.JAVA_VERSION=="JAVA17"){
+		return "/home/builder/java/jdk-17.0.2"
+	}else if(env.JAVA_VERSION=="JAVA11"){
+		return "/home/builder/java/jdk-11.0.2"
+	}else if(env.JAVA_VERSION=="JAVA21"){
 		return "/home/builder/java/jdk-21.0.1"
 	}else{
-        return "/home/builder/java/openjdk-1.8.0-262"
-    }
+		return "/home/builder/java/openjdk-1.8.0-262"
+	}
 }
 
-def runAllTests(String type, String version, Boolean useReverseProxy){
-            copyRPM type, version
-            sh 'sudo /usr/local/sbin/mladmin removeforest /space/Forests'
-            setUpML '$WORKSPACE/xdmp/src/Mark*.rpm'
-            copyConvertersRPM type,version
-            setUpMLConverters '$WORKSPACE/xdmp/src/Mark*Converters*.rpm'
+def setupDockerMarkLogic(String image){
+	sh label:'mlsetup', script: '''#!/bin/bash
+	echo "Removing any running MarkLogic server and clean up MarkLogic data directory"
+    sudo /usr/local/sbin/mladmin remove
+    sudo /usr/local/sbin/mladmin cleandata
+    cd java-client-api/test-app
+    mkdir -p docker/marklogic/logs
+    docker compose down -v || true
+    echo "Using image: "'''+image+'''
+    MARKLOGIC_IMAGE='''+image+''' docker compose up -d --build
+	echo "mlPassword=admin" > gradle-local.properties
+    echo "Waiting for MarkLogic server to initialize."
+    sleep 30s
+    cd ..
+	export GRADLE_USER_HOME=$WORKSPACE/$GRADLE_DIR
+	export PATH=$GRADLE_USER_HOME:$JAVA_HOME/bin:$PATH
+   	./gradlew -i mlDeploy mlReloadSchemas
+  '''
+}
 
-            sh label:'deploy test app', script: '''#!/bin/bash
-                export JAVA_HOME=$JAVA_HOME_DIR
-                export GRADLE_USER_HOME=$WORKSPACE/$GRADLE_DIR
-                export PATH=$GRADLE_USER_HOME:$JAVA_HOME/bin:$PATH
-                cd java-client-api
-                ./gradlew -i mlDeploy mlReloadSchemas -PmlForestDataDirectory=/space
-            '''
+def runAllTests(Boolean useReverseProxy, String image){
```

Comment:
Not important now, but a nice refactor (which would have been useful before this PR too) would to make two separate functions:

- `runTests(image)`
- `runTestsWithReverseProxy(image)`

That gets rid of any confusion below over what the boolean refers to, and what position it should be in the arg list. 

Again, not important now, but a good refactor after this PR.